activation/interface.go

```diff
@@ -61,7 +62,7 @@ type layerClock interface {
 }
 
 type nipostBuilder interface {
-	BuildNIPost(ctx context.Context, challenge *types.NIPostChallenge) (*types.NIPost, error)
+	BuildNIPost(ctx context.Context, challenge *types.NIPostChallenge, certifier certifierService) (*types.NIPost, error)
```

Comment:
This comment still applies

Reply:
The problem with the approach you suggest is that the PoST, which is the crucial data for certifier is not available when the atx and nipost builders are constructed. That's why the certifier is created inside the atxBuilder, after the first PoST is ready.

Reply:
Yeah but this I thought that this will only be one of multiple possible ways to get a certificate to be used for a PoET?

Instead the certifier could just return a typed error `ErrMissingPost` or similar when the nipost builder requests a certificate for a poet and the certifier is unable to request that due to a missing post.

The initial post is always created and stored in the database before the certificate is requested anyway so theoretically this error should never even be needed to be returned. And the logic to fetch a post from an old ATX for the given identity should in my opinion also be part of the certifier rather than the nipost builder (and only after it doesn't find any post for the identity return an error).